pkg/updatestatus/nodeinformer.go

```diff
@@ -155,6 +165,52 @@ func (c *nodeInformerController) sync(ctx context.Context, syncCtx factory.SyncC
 	return nil
 }
 
+type machineConfigVersionCache struct {
+	cache map[string]string
+	lock  sync.Mutex
+}
+
+func (c *machineConfigVersionCache) ingest(mc *machineconfigv1.MachineConfig) bool {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	v, ok := c.cache[mc.Name]
+	if openshiftVersion, exist := mc.Annotations[mco.ReleaseImageVersionAnnotationKey]; exist && openshiftVersion != "" {
+		if !ok || v != openshiftVersion {
+			if c.cache == nil {
+				c.cache = make(map[string]string)
+			}
+			c.cache[mc.Name] = openshiftVersion
+			klog.V(4).Infof("Cached MachineConfig %s with version %s", mc.Name, openshiftVersion)
+			return true
+		}
+	} else if ok {
+		delete(c.cache, mc.Name)
+		klog.V(4).Infof("Deleted MachineConfig %s from the cache as no version can be found", mc.Name)
+		return true
+	}
+
+	return false
+}
+
+func (c *machineConfigVersionCache) forget(name string) bool {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if _, ok := c.cache[name]; ok {
+		delete(c.cache, name)
+		klog.V(4).Infof("Deleted MachineConfig %s from the cache", name)
+		return true
+	}
+	return false
+}
+
+func (c *machineConfigVersionCache) match(config string) (string, bool) {
```

Comment:
nit: consider `versionFor(name string)`

tells more directly what is the output, and param `name` is consistent with `forget`

Reply:
Done.

https://github.com/openshift/cluster-version-operator/pull/1164/commits/a81ec69e31f504ce10a44650cfa5c4d4260fd2cd#diff-c5a41cb83200e13544e0509a5e77c180e16a1d70d7411a79cac2bc107e89950fR203